examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -28,144 +25,378 @@
+       to the spectral ratio
+       See [1] Nikulin 2011, Eq. (24)
+
+   return_filtered : bool (default False)
+        If return_filtered is True, data is bandpassed and projected onto
+        the SSD components.
+
+   n_fft: int (default None)
+       if sort_by_spectral_ratio is set to True, then the sources will be
+       sorted accordinly to their spectral ratio which is calculated based on
+       "psd_array_welch" function. the n_fft set the length of FFT used.
+       See mne.time_frequency.psd_array_welch for more information
+
+   cov_method_params : TYPE, optional
+        As in mne.decoding.SPoC
+        The default is None.
+   rank : None | dict | ‘info’ | ‘full’
+        As in mne.decoding.SPoC
+        This controls the rank computation that can be read from the
+        measurement info or estimated from the data.
+        See Notes of mne.compute_rank() for details.The default is None.
+        The default is None.
+
+    REFERENCES:
+    [1] Nikulin, V. V., Nolte, G., & Curio, G. (2011). A novel method for
+    reliable and fast extraction of neuronal EEG/MEG oscillations on the basis
+    of spatio-spectral decomposition. NeuroImage, 55(4), 1528-1535.
+    [2] Haufe, S., Dähne, S., & Nikulin, V. V. (2014). Dimensionality reduction
+    for the analysis of brain oscillations. NeuroImage, 101, 583-597.
     """
 
-    def __init__(self, filt_params_signal, filt_params_noise,
-                 estimator='oas', n_components=None,
-                 subtract_signal=True,
-                 reject=None,
-                 flat=None,
-                 picks=None,
-                 sort_by_spectral_ratio=True):
+    def __init__(self, filt_params_signal, filt_params_noise, sampling_freq,
+                 estimator='oas', n_components=None, picks=None,
+                 sort_by_spectral_ratio=True, return_filtered=False,
+                 n_fft=None, cov_method_params=None, rank=None):
         """Initialize instance"""
 
         dicts = {"signal": filt_params_signal, "noise": filt_params_noise}
         for param, dd in [('l', 0), ('h', 0), ('l', 1), ('h', 1)]:
             key = ('signal', 'noise')[dd]
-            if  param + '_freq' not in dicts[key]:
+            if param + '_freq' not in dicts[key]:
                 raise ValueError(
-                    "'%' must be defined in filter parameters for %s" % key)
+                    "'%%' must be defined in filter parameters for %s" % key)
             val = dicts[key][param + '_freq']
             if not isinstance(val, (int, float)):
                 raise ValueError(
                     "Frequencies must be numbers, got %s" % type(val))
+        # check freq bands
+        if (filt_params_noise['l_freq'] > filt_params_signal['l_freq'] or
+                filt_params_signal['h_freq'] > filt_params_noise['h_freq']):
+            raise ValueError('Wrongly specified frequency bands!\n \
+                The signal band-pass must be within the t noise band-pass!')
```

Comment:
\ is not needed here as you have surrounding round brackets.

Reply:
I got an error if I delete it in my python interpreter (spyder)

Reply:
Ahh I was not looking properly. You can try something like this:

```Pyrhon
raise ValueError('Wrongly specified frequency bands!\n'
                            'The signal band-pass must be within the t noise band-pass!')
```